codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/protocols/ServerHttpBoundProtocolGenerator.kt

```diff
@@ -207,9 +209,31 @@ private class ServerHttpBoundProtocolTraitImplGenerator(
                         )
                 }
             }
+
+            impl<B> #{SmithyHttpServer}::request::FromRequest<#{Marker}, B> for #{I}
+            where
+                B: #{SmithyHttpServer}::body::HttpBody + Send,
+                B: 'static,
+                ${streamingBodyTraitBounds(operationShape)}
+                B::Data: Send,
+                #{RequestRejection} : From<<B as #{SmithyHttpServer}::body::HttpBody>::Error>
+            {
+                type Rejection = #{RuntimeError};
+                type Future = std::pin::Pin<Box<dyn std::future::Future<Output = Result<Self, Self::Rejection>> + Send>>;
```

Comment:
I guess we should define `Pin`, `Box`, `Future` in `RuntimeTypes.kt`

Reply:
Side note: I hope that this associated type will not exist as a `Pin<Box<_>>` for long - when we are landing https://github.com/awslabs/smithy-rs/issues/1706 we should be able to give this a concrete type.